### PR TITLE
Replace Galaxy S20 with Pixel 6 for Android testing

### PR DESCRIPTION
**Pull request by antiagainst**
We dropped S20 for benchmarking. Better to be
consistent here for testing.

Also perform artifacts clean up before each step.

### CODE REVIEW

**Comment by antiagainst** [comment]
Triggered build: https://buildkite.com/iree/iree-android-arm64-v8a/builds/5552

**Review by GMNGeoffrey — APPROVED** [review]

File: build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml [hunk 1]
```diff
@@ -18,22 +18,24 @@ steps:
 
   - wait
 
-  - label: "test on Galaxy S20 (exynos-990, mali-g77)"
+  - label: "test on Pixel 6 (google-tensor, mali-g78)"
     commands:
+      - "git clean -fdx"
```
Comment (ScottTodd) [inline build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml:23]: what's this for? seems unrelated to changing which phone is used?

**Review by ScottTodd — APPROVED** [review]

File: build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml (hunk 1 above)
Comment (antiagainst) [reply build_tools/buildkite/cmake/android/arm64-v8a/pipeline.yml:23]: It's cleaning up previously downloaded artifacts on RPIs to avoid seeing out of storage issue. Yes not related to phone change. But trivial enough to bundle (as I don't think we will have issues with this change so need separate reverting). I added a note in the commit message.
